src/utils/permit.ts

```diff
@@ -47,10 +47,11 @@ export async function createPermitSignature(
   tokenContract: ERC20Permit,
 ): Promise<ERC20PermitSignatureStruct> {
   const signerAddress = await signer.getAddress();
-  const [nonce, tokenAddress, tokenName] = await Promise.all([
+  const [nonce, tokenAddress, tokenName, decimals] = await Promise.all([
     tokenContract.nonces(signerAddress),
     tokenContract.getAddress(),
     tokenContract.name(),
+    tokenContract.decimals(),
```

Comment:
Maybe we can add a `token` parameter here and fetch the decimals from the config (and avoid using another RPC call).